web/src/composebox_typeahead.js

```diff
@@ -820,6 +821,21 @@ export function content_typeahead_selected(item, event) {
                     is_silent,
                 );
                 beginning += mention_text + " ";
+                const compose_stream_id = stream_data.get_stream_id(compose_state.stream_name());
+                const edit_stream_id = $textbox
+                    .closest(".recipient_row")
+                    .find(".message_header_stream")
+                    .attr("data-stream-id");
+                const stream_id = Number(compose_stream_id || edit_stream_id);
```

Comment:
This logic doesn't make any sense. If we're typing in the compose box, we need to use the compose box's stream ID. Otherwise we use the edit box stream ID. Under no circumstances does it make sense to use the compose box stream ID if we're editing a message; that is introducing another instance of #25410.

Reply:
Got it! Thanks so much for the explanation.